Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp

```diff
@@ -806,6 +848,42 @@ void ezDebugRenderer::Draw3DText(const ezDebugRendererContext& context, const ez
   });
 }
 
+void ezDebugRenderer::AddPersistentCross(const ezDebugRendererContext& context, float fSize, const ezColor& color, const ezTransform& transform, ezTime duration)
+{
+  EZ_LOCK(s_Mutex);
```

Comment:
Maybe use a separate mutex for persistent stuff to prevent contention.

Reply:
I considered that, but in ezDebugRenderer::RenderInternal we call all the "DrawCross" etc functions, which again go through this mutex, so I think it's more efficient to get that lock once, instead of multiple times. And then having a second mutex doesn't gain you anything.